src/libPOW/pow.cpp

```diff
@@ -474,4 +449,59 @@ ethash_return_value_t POW::LightHash(uint64_t blockNum,
 {
     EthashConfigureLightClient(blockNum);
     return EthashLightCompute(ethash_light_client, header_hash, nonce);
+}
+
+void POW::InitOpenCL()
+{
+    using namespace dev::eth;
+    CLMiner::setCLKernel(CLKernelName::Stable);
+
+    ptree pt;
+    read_xml("constants.xml", pt);
+    std::string parentNode = "node.gpu.opencl.";
+    auto localWorkSize = pt.get<unsigned int>(parentNode + "LOCAL_WORK_SIZE");
+    auto globalWorkSizeMultiplier
+        = pt.get<unsigned int>(parentNode + "GLOBAL_WORK_SIZE_MULTIPLIER");
+    auto startEpoch = pt.get<unsigned int>(parentNode + "START_EPOCH");
+
+    if (!CLMiner::configureGPU(localWorkSize, globalWorkSizeMultiplier, 0,
+                               startEpoch, 0, 0, false, false))
+    {
+        throw std::runtime_error(
+            "Failed to configure OpenCL GPU, please check hardware");
+    }
+
+    CLMiner::setNumInstances(UINT_MAX);
+    m_miner = std::make_unique<CLMiner>();
+    LOG_GENERAL(INFO, "OpenCL GPU initialized in POW");
+}
+
+void POW::InitCUDA()
+{
+#ifdef CUDA_MINE
+    using namespace dev::eth;
+
+    ptree pt;
+    read_xml("constants.xml", pt);
```

Comment:
Since we are adding the parameters to `constants.xml`, can we move these lines of reading configuration to `constants.h/cpp`? It will reduce the difficulty of refactoring the configuration loading in the future.